crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -444,6 +448,50 @@ pub trait WorldQueryGats<'world> {
     type Fetch;
 }
 
+/// An extension of [`WorldQuery`] for batched queries.
+///
+pub trait WorldQueryBatch<const N: usize, const ALIGN: usize>:
+    for<'w> WorldQueryBatchGats<'w, N, ALIGN> + for<'w> WorldQueryGats<'w>
+{
+    /// Retrieve a batch of size `N` with desired alignment `ALIGN` from the current table.
```

Comment:
This need a little more detail on how to select (and verify) the correct alignment.

Reply:
Agreed, but I'm not sure where the best place for that documentation should be.  In general the alignment you'd choose would be the greatest alignment possible for your chosen batch.  There's a compile time assertion preventing you from getting it wrong, fortunately :) the expectation is that in the future when "generic const expressions" are stable, you won't need to type this at all anymore since it'll be computed for you (the batch alignment is actually just `gcd(MAX_SIMD_ALIGNMENT, N * size_of::<T>())`).  Having to enter the alignment manually is just a stopgap measure until that's stable.

As an example for how to choose `ALIGN`, consider a batch of 4 `Vec3`s.  A single `Vec3` is 12 bytes and a batch of 4 is 48 bytes.   Since all `BlobVec`s start on `MAX_SIMD_ALIGNMENT` boundaries (64 bytes), this means that every batch will start on a 16 byte boundary guaranteed: the linear combination of `blob_vec_addr + i*48` will always be divisible by 16.  In practice this means we can just check whether `ALIGN` divides `N * size_of::<T>()`.

To make it easy and safe, a compile time assertion will check that `ALIGN` divides `N * size_of::<T>()` and fail the build if it doesn't.  This is the safest option as it would lead to UB otherwise or require runtime checks.

Reply:
I added a little more detail on this in one of my latest commits and placed it in `for_each_mut_batched`.  With the recent decision to wait until GAT stabilization, more detail might not be needed, as it will be automatically derived :) is it OK if I resolve this, in light of that?

Reply:
GATs is being stabilized not generic const exprs

Reply:
In that case it will still be necessary to manually specify `ALIGN`.  I'll see what can be done though.